app/components/channel_info/index.js

```diff
@@ -0,0 +1,420 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {injectIntl, intlShape} from 'react-intl';
+import {
+    Dimensions,
+    Platform,
+    TouchableWithoutFeedback,
+    View,
+    Text,
+    findNodeHandle
+} from 'react-native';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import Loading from 'app/components/loading';
+import StatusBar from 'app/components/status_bar';
+import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {getShortenedURL} from 'app/utils/url';
+
+class ChannelInfo extends PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        navigator: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        displayHeaderOnly: PropTypes.bool,
+        creating: PropTypes.bool,
+        editing: PropTypes.bool,
+        updating: PropTypes.bool,
+        error: PropTypes.string,
+        displayName: PropTypes.string,
+        currentTeamUrl: PropTypes.string,
+        channelURL: PropTypes.string,
+        purpose: PropTypes.string,
+        header: PropTypes.string,
+        enableRightButton: PropTypes.func,
+        onDisplayNameChange: PropTypes.func,
+        onChannelURLChange: PropTypes.func,
+        onPurposeChange: PropTypes.func,
+        onHeaderChange: PropTypes.func,
+        oldChannel: PropTypes.shape({
+            displayName: PropTypes.string,
+            currentTeamUrl: PropTypes.string,
+            channelURL: PropTypes.string,
+            purpose: PropTypes.string,
+            header: PropTypes.string
+        }),
+        actions: PropTypes.shape({
+            handleCreateChannel: PropTypes.func,
+            patchChannel: PropTypes.func
+        })
+    };
+
+    static defaultProps = {
+        editing: false
+    };
+
+    blur = () => {
+        this.nameInput.refs.wrappedInstance.blur();
```

Comment:
can we make sure the refs here are defined to avoid crashes?

Reply:
@enahum do you guys prefer

`this.nameInput && this.nameInput.refs.wrappedInstance.blur();`

or

```
if (this.nameInput) {
   this.nameInput.refs.wrappedInstance.blur();
}
```

Reply:
the second one